vip-integration-helper/integrations/integration.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace Automattic\VIP\Integrations;
+
+defined( 'ABSPATH' ) || die();
+
+/**
+ * Abstract base class for all integration implementations.
+ */
+abstract class Integration {
+	/**
+	 * An optional configuration array for this integration, added during activation.
+	 *
+	 * @var array
+	 */
+	public $config = [];
+
+	/**
+	 * A boolean indicating if this integration should be loaded. Defaults to false.
+	 *
+	 * @var bool
+	 */
+	protected $active = false;
+
+	/**
+	 * Activates this integration with an optional configuration value.
+	 *
+	 * @param array  $config An associative array of configuration values for the integration.
+	 */
+	public function activate( $config = [] ): void {
```

Comment:
If this is documented to be an array, then where's the type validation?

```suggestion
	public function activate( array $config = [] ): void {
```

Of course, it limits being able to use ArrayObject, or a custom configuration class per each integration, but the rest of the logic/docs makes assumptions that `$config` is expected to be an array anyway.

Reply:
Actually, why is the configuration being set at activation time? Is it likely to change from the registration (`integrate`) time? Is this to allow customer-overrides of the config?

Reply:
> Actually, why is the configuration being set at activation time?
> ...
> Is this to allow customer-overrides of the config?

Yep, that's the idea. If we'd like to provide user-facing integration config, we can provide options like:

```php
\Automattic\VIP\Integrations\activate( 'some-integration', [
    'API_KEY' => '974f2acb...',
] );
```

These provide a way to connect configuration in the same place as activation, rather than using constants or something in addition.